ladybug/windrose.py

```diff
@@ -0,0 +1,953 @@
+
+    @property
+    def inner_boundary_circle(self):
+        """Get an Arc2D for the inner boundary of the plot."""
+        return self._compass.inner_boundary_circle
+
+    @property
+    def orientation_lines(self):
+        """Orientation lines for windrose as a LineSegment2D list."""
+        lines = WindRose._xtick_radial_lines(self._bin_vectors)
+        return [line.scale(self._radius).move(self._center_point2d) for line in lines]
+
+    @property
+    def orientation_labels(self):
+        """List of labels corresponding to the orientation angles."""
+        if self._orientation_labels is None:
+            self._orientation_label = self.angles
+        return self._orientation_labels
+
+    @property
+    def orientation_label_points(self):
+        """Get a list of label points from a list of orientation angles between 0 and 360.
+        """
+        return self._compass.label_points_from_angles(
+            self.angles, factor=self.label_point_size_factor)
+
+    @property
+    def orientation_label_ticks(self):
+        """Get a list of Linesegment2Ds from a list of orientation angles between 0 and 360."""
+        return self._compass.ticks_from_angles(self.angles,
+                                               factor=self.label_tick_size_factor)
+
+    @property
+    def frequency_lines(self):
+        """Frequency lines for windrose as Polygon2D lists."""
+        freqs = WindRose._ytick_radial_lines(self._bin_vectors,
+                                             self.legend_parameters.segment_count)
+        return [freq.scale(self._radius).move(self._center_point2d) for freq in freqs]
+
+    @property
+    def _center_point2d(self):
+        """Compute center of windrose plot as Point2D."""
+        return Point2D(self.base_point.x + self._radius,
+                       self.base_point.y + self._radius)
+
+    @property
+    def _radius(self):
+        """Radius of the inner boundary of the windrose plot.
+
+        This number will be a integer multiple of the legend parameter segement count.
+        """
+        ytick_num = self.legend_parameters.segment_count
+        if self.show_zeros:
+            ytick_num += self._ytick_zero_inc
+        return self.frequency_spacing_distance * ytick_num
+
+    @property
+    def _wind_radius(self):
+        """Radius of the windrose mesh."""
+        if self.show_zeros:
+            return (self.real_freq_max + self.zeros_per_bin) / \
+                (self.frequency_maximum + self.zeros_per_bin) * self._radius
+        else:
+            return self.real_freq_max / self.frequency_maximum * self._radius
+
+    @property
+    def _ytick_zero_inc(self):
+        """Returns how much yticks increasa with zeros"""
+        return ceil(self.zeros_per_bin / (self.real_freq_max + self.zeros_per_bin) *
+                    self.legend_parameters.segment_count)
+
+    @property
+    def _compass(self):
+        """Compass object for the windrose plot."""
```

Comment:
This should be public and then we can get rid of all the other properties that duplicate what the compass does.

Reply:
Since the basepoint, radius, and legend_parameters segment count all impact the way this object, we will have to set this property to None if any of those properties chagne. I'll add this as a note for future coders. As discussed, I think this might get messy as we scale up if we're mutating properties invisibly in other properties. I think the best way to handle this is just to cache these properties based on function arguements (memoization) which allows us to cache different instances of these classes based on changing properties without having to mutate it in other properties.